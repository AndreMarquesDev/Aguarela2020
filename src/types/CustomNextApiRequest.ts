import type { NextApiRequest } from 'next';
import type { FormPostRequestBody } from './FormPostRequestBody';

export type CustomNextApiRequest = {
    body: FormPostRequestBody;
} & NextApiRequest;
