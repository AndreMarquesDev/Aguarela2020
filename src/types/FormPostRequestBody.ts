import type { FormValues } from './FormValues';

export type FormPostRequestBody = {
    isPlaywright: boolean;
} & FormValues;
