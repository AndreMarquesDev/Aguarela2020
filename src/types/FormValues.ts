export type FormValues = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};
