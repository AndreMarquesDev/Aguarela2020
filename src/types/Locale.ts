export enum Locale {
    Pt = 'pt',
    En = 'en',
}
