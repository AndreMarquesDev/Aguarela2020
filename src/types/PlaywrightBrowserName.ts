export enum PlaywrightBrowserName {
    Chromium = 'chromium',
    Firefox = 'firefox',
    Webkit = 'webkit',
}
