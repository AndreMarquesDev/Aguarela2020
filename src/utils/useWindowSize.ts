import { useEffect, useState } from 'react';

type WindowSize = {
    width: number;
    height: number;
};

export const useWindowSize = (): WindowSize => {
    const [windowSize, setWindowSize] = useState<WindowSize>({
        width: 0,
        height: 0,
    });

    useEffect(() => {
        const handleResize = (): void => {
            // eslint-disable-next-line react-hooks-extra/no-direct-set-state-in-use-effect
            setWindowSize({
                width: window.innerWidth,
                height: window.innerHeight,
            });
        };

        window.addEventListener('resize', handleResize);

        handleResize();

        return (): void => window.removeEventListener('resize', handleResize);
    }, []);

    return windowSize;
};

export enum Breakpoint {
    Desktop = 1280,
    Phablet = 1023,
    Tablet = 767,
    Mobile = 511,
}
