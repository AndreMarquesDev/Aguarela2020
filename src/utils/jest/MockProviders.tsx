/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
