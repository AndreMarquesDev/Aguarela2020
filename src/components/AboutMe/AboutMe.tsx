import React, { useContext } from 'react';
import Image from 'next/image';
import { Button } from '../Button/Button';
import { TextsContext } from '../context/TextsContext';
import { Title } from '../Title/Title';
import { useWindowSize, Breakpoint } from '../../utils/useWindowSize';
import { aboutMeSectionDataTestId } from '../../utils/dataTestIds';
import { catarinaSantiagoInstagramUrl } from '../../utils/urls';
import { desktopMediaQuery, mobileMediaQuery, tabletMediaQuery } from '../../styles/mediaQueries';

export const AboutMe = (): JSX.Element => {
    const { texts } = useContext(TextsContext);
    const windowSize = useWindowSize();
    const isDesktop = windowSize.width > Breakpoint.Desktop;

    return (
        <>
            <section className="container" data-testid={aboutMeSectionDataTestId}>
                <div className="wrapper genericMargins">
                    <Title colored text={texts.about} />
                    <div className="mainContentContainer">
                        <div className="imageBlock">
                            <Image
                                priority
                                alt="a picture of me, Catarina"
                                height={650}
                                sizes={`(${desktopMediaQuery}) 45vw, (${tabletMediaQuery}) 75vw, (${mobileMediaQuery}) 100vw, 35vw`}
                                src="/images/about-me.jpg"
                                style={{
                                    width: '100%',
                                    height: 'auto',
                                    position: 'relative',
                                }}
                                width={530}
                            />
                        </div>
                        <div className="textBlock">
                            <p className="bodyText">{texts.hiMyNameIs}</p>
                            <p className="bodyText">{texts.withMoreThan}</p>
                            <p className="bodyText">{texts.theSocialMediaCommunicationStrategy}</p>
                            <Button
                                alignLeft={isDesktop}
                                externalLink={catarinaSantiagoInstagramUrl}
                            >
                                {texts.getToKnowMeBetter}
                            </Button>
                        </div>
                    </div>
                </div>
            </section>

            <style jsx>
                {`
                    .container {
                        background: var(--white);

                        .mainContentContainer {
                            display: flex;
                            justify-content: space-between;
                            align-items: center;

                            @media (${desktopMediaQuery}) {
                                flex-direction: column;
                                justify-content: center;
                            }
                        }

                        .imageBlock {
                            width: 35%;
                            position: relative;

                            &:before {
                                content: '';
                                width: 100%;
                                height: 100%;
                                position: absolute;
                                top: 12rem;
                                left: 12rem;
                                background: var(--pink);
                            }

                            @media (${desktopMediaQuery}) {
                                width: 45%;
                                margin-bottom: 50rem;
                            }

                            @media (${tabletMediaQuery}) {
                                width: 75%;
                            }

                            @media (${mobileMediaQuery}) {
                                width: 100%;
                            }
                        }

                        .textBlock {
                            width: 60%;

                            p {
                                text-align: left;
                            }

                            @media (${desktopMediaQuery}) {
                                width: 80%;

                                p {
                                    text-align: center;
                                }
                            }

                            @media (${tabletMediaQuery}) {
                                width: 100%;
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};
