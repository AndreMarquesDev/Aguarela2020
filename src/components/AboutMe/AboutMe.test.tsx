import type { RenderResult } from '@testing-library/react';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Locale } from '../../types/Locale';
import { MockProviders } from '../../utils/jest/MockProviders';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';
import { textsEn, textsPt } from '../../utils/texts';
import { Breakpoint } from '../../utils/useWindowSize';
import { AboutMe } from './AboutMe';
import '@testing-library/jest-dom';

const renderComponent = (language: Locale = Locale.Pt): RenderResult => {
    return render(
        <MockProviders language={language}>
            <AboutMe />
        </MockProviders>,
    );
};

describe('<AboutMe />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        expect(screen.getByText(textsPt.about)).toBeInTheDocument();
        expect(screen.getByText(textsPt.hiMyNameIs)).toBeInTheDocument();
        expect(screen.getByText(textsPt.withMoreThan)).toBeInTheDocument();
        expect(screen.getByText(textsPt.theSocialMediaCommunicationStrategy)).toBeInTheDocument();
        expect(screen.getByText(textsPt.getToKnowMeBetter)).toBeInTheDocument();

        // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
        expect(container.getElementsByClassName('alignCenter').length).toBe(0);
        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = renderComponent(Locale.En);

        expect(screen.getByText(textsEn.about)).toBeInTheDocument();
        expect(screen.getByText(textsEn.hiMyNameIs)).toBeInTheDocument();
        expect(screen.getByText(textsEn.withMoreThan)).toBeInTheDocument();
        expect(screen.getByText(textsEn.theSocialMediaCommunicationStrategy)).toBeInTheDocument();
        expect(screen.getByText(textsEn.getToKnowMeBetter)).toBeInTheDocument();

        // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
        expect(container.getElementsByClassName('alignCenter').length).toBe(0);
        expect(container).toMatchSnapshot();
    });

    test('renders properly in mobile', () => {
        setJestWindowWidth(Breakpoint.Mobile);

        const { container } = renderComponent();

        // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
        expect(container.getElementsByClassName('alignCenter').length).toBe(1);
        expect(container).toMatchSnapshot();
    });
});
