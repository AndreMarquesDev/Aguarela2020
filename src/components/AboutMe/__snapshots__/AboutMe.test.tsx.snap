// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutMe /> renders properly 1`] = `
<div>
  <section
    class="jsx-2878702874 container"
    data-testid="aboutMe_section"
  >
    <div
      class="jsx-2878702874 wrapper genericMargins"
    >
      <h1
        class="jsx-1254015016 fontL colored"
      >
        Sobre
      </h1>
      <div
        class="jsx-2878702874 mainContentContainer"
      >
        <div
          class="jsx-2878702874 imageBlock"
        >
          <img
            alt="a picture of me, Catarina"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="650"
            src="/_next/image?url=%2Fimages%2Fabout-me.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimages%2Fabout-me.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fabout-me.jpg&w=1080&q=75 2x"
            style="color: transparent; width: 100%; height: auto; position: relative;"
            width="530"
          />
        </div>
        <div
          class="jsx-2878702874 textBlock"
        >
          <p
            class="jsx-2878702874 bodyText"
          >
            Olá! O meu nome é Catarina Santiago e sou gestora de redes sociais e fotógrafa baseada em Lisboa.
          </p>
          <p
            class="jsx-2878702874 bodyText"
          >
            Com mais de 4 anos e 18 clientes de experiência, um projeto de freelance transformou-se na Aguarela Digital, uma agência de comunicação com o objetivo de posicionar e fazer brilhar marcas e negócios no digital.
          </p>
          <p
            class="jsx-2878702874 bodyText"
          >
            A estratégia de comunicação em redes sociais é sempre desenvolvida com base nos objetivos do cliente. Embora seja importante ser flexível e compreensiva, o meu papel é pensar fora da caixa, criar conteúdo inovador para redes sociais e otimizar a estratégia com base nos resultados e tendências de mercado. Acredito que uma boa estratégia de comunicação aliada a um planeamento visual de conteúdo é o segredo para o aumento da visibilidade e faturação do negócio.
          </p>
          <div
            class="jsx-477676945 buttonWrapper"
          >
            <div
              class="jsx-477676945 buttonBackground"
            >
              <a
                class="jsx-477676945 buttonComponent fontS"
                href="https://www.instagram.com/catarinasantiago/"
                rel="noreferrer"
                target="_blank"
              >
                Conhece-me melhor
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<AboutMe /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-2878702874 container"
    data-testid="aboutMe_section"
  >
    <div
      class="jsx-2878702874 wrapper genericMargins"
    >
      <h1
        class="jsx-1254015016 fontL colored"
      >
        About
      </h1>
      <div
        class="jsx-2878702874 mainContentContainer"
      >
        <div
          class="jsx-2878702874 imageBlock"
        >
          <img
            alt="a picture of me, Catarina"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="650"
            src="/_next/image?url=%2Fimages%2Fabout-me.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimages%2Fabout-me.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fabout-me.jpg&w=1080&q=75 2x"
            style="color: transparent; width: 100%; height: auto; position: relative;"
            width="530"
          />
        </div>
        <div
          class="jsx-2878702874 textBlock"
        >
          <p
            class="jsx-2878702874 bodyText"
          >
            Hi! My name is Catarina Santiago and I'm a social media manager and photographer based in Lisbon.
          </p>
          <p
            class="jsx-2878702874 bodyText"
          >
            With more than 4 years of experience and 18 clients, a freelance project became Aguarela Digital, a communication agency with the goal to position and make brands and businesses shine in the digital world.
          </p>
          <p
            class="jsx-2878702874 bodyText"
          >
            The social media strategy is always developed based on the client's goals. While it is important to be flexible, my role is to think outside the box by creating innovative content and optimizing the strategy based on results and market trends. I believe that a good communication strategy combined with a visual planning of the content is the key to increase the visibility and revenue of the business.
          </p>
          <div
            class="jsx-477676945 buttonWrapper"
          >
            <div
              class="jsx-477676945 buttonBackground"
            >
              <a
                class="jsx-477676945 buttonComponent fontS"
                href="https://www.instagram.com/catarinasantiago/"
                rel="noreferrer"
                target="_blank"
              >
                Get to know me better
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<AboutMe /> renders properly in mobile 1`] = `
<div>
  <section
    class="jsx-2878702874 container"
    data-testid="aboutMe_section"
  >
    <div
      class="jsx-2878702874 wrapper genericMargins"
    >
      <h1
        class="jsx-1254015016 fontL colored"
      >
        Sobre
      </h1>
      <div
        class="jsx-2878702874 mainContentContainer"
      >
        <div
          class="jsx-2878702874 imageBlock"
        >
          <img
            alt="a picture of me, Catarina"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="650"
            src="/_next/image?url=%2Fimages%2Fabout-me.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimages%2Fabout-me.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fabout-me.jpg&w=1080&q=75 2x"
            style="color: transparent; width: 100%; height: auto; position: relative;"
            width="530"
          />
        </div>
        <div
          class="jsx-2878702874 textBlock"
        >
          <p
            class="jsx-2878702874 bodyText"
          >
            Olá! O meu nome é Catarina Santiago e sou gestora de redes sociais e fotógrafa baseada em Lisboa.
          </p>
          <p
            class="jsx-2878702874 bodyText"
          >
            Com mais de 4 anos e 18 clientes de experiência, um projeto de freelance transformou-se na Aguarela Digital, uma agência de comunicação com o objetivo de posicionar e fazer brilhar marcas e negócios no digital.
          </p>
          <p
            class="jsx-2878702874 bodyText"
          >
            A estratégia de comunicação em redes sociais é sempre desenvolvida com base nos objetivos do cliente. Embora seja importante ser flexível e compreensiva, o meu papel é pensar fora da caixa, criar conteúdo inovador para redes sociais e otimizar a estratégia com base nos resultados e tendências de mercado. Acredito que uma boa estratégia de comunicação aliada a um planeamento visual de conteúdo é o segredo para o aumento da visibilidade e faturação do negócio.
          </p>
          <div
            class="jsx-477676945 buttonWrapper"
          >
            <div
              class="jsx-477676945 buttonBackground alignCenter"
            >
              <a
                class="jsx-477676945 buttonComponent fontS"
                href="https://www.instagram.com/catarinasantiago/"
                rel="noreferrer"
                target="_blank"
              >
                Conhece-me melhor
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
